src/vhost_kern/vdpa.rs

```diff
@@ -0,0 +1,375 @@
+// Copyright (C) 2021 Red Hat, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0 or BSD-3-Clause
+
+//! Kernel-based vhost-vdpa backend.
+
+use std::fs::{File, OpenOptions};
+use std::os::raw::c_int;
+use std::os::unix::fs::OpenOptionsExt;
+use std::os::unix::io::{AsRawFd, RawFd};
+
+use vm_memory::GuestAddressSpace;
+use vmm_sys_util::eventfd::EventFd;
+use vmm_sys_util::ioctl::{ioctl_with_mut_ref, ioctl_with_ptr, ioctl_with_ref};
+
+use std::alloc::{alloc, dealloc, Layout};
+use std::mem;
```

Comment:
Group all std dependencies together by moving it near to line 9?

Reply:
Yep, I'll fix.

Reply:
It would also be great to give a simple introduction about vDPA as the module documentation.

Reply:
yep sure, I'll add an introduction!